### ISSUE TITLE
Application crash on second log roll over

### ISSUE THREAD

**Issue by mmatt1967** [kohsuke/winsw#314]
When winsw is configured for "roll-by-size-time" logging, the application crashes on the second size based roll over. For example, when the logging config below is used, the application will crash as soon as the log file reaches 10 MB for the second time in a day.

 ```
 <log mode="roll-by-size-time">
    <sizeThreshold>10240</sizeThreshold>
    <pattern>yyyyMMdd</pattern>
    <autoRollAtTime>00:00:00</autoRollAtTime>
    <zipOlderThanNumDays>1</zipOlderThanNumDays>
    <zipDateFormat>yyyyMMdd</zipDateFormat>
  </log>
```

A series of errors like the following is logged to Windows Application Event Log:
```
 1) Failed to roll size time log: Failed to process file ...\logs\AppName.20190426.#0001.out.log due to error File ...\logs\AppName.20190426.#0001.out.log does not follow the pattern provided

 2)Application: AppName.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ObjectDisposedException
   at System.IO.__Error.FileNotOpen()
   at System.IO.FileStream.Flush(Boolean)
   at winsw.RollingSizeTimeLogAppender.CopyStreamWithRotation(System.IO.Stream, System.String)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart()

3)Faulting application name: AppName.exe, version: 2.2.0.0, time stamp: 0x5c325f61
Faulting module name: KERNELBASE.dll, version: 6.2.9200.22673, time stamp: 0x5c5df37d
Exception code: 0xe0434352
Fault offset: 0x0000000000027518
Faulting process id: 0x694
Faulting application start time: 0x01d4fc39de96d677
Faulting application path: ...\AppName.exe
Faulting module path: C:\Windows\system32\KERNELBASE.dll
Report Id: 88fdf989-6847-11e9-9411-00155d3c6663
Faulting package full name: 
Faulting package-relative application ID: 

```

**Comment by NextTurn** [comment]
Please close this. This has been fixed in #315.

**Comment by NextTurn** [comment]
This has been fixed in #315.

**Comment by oleg-nenashev** [comment]
Sorry, missed that
